### ISSUE TITLE
Support marking a route traveled without specifying a whole segment

### ISSUE THREAD

**Issue by jteresco** [TravelMapping/DataProcessing#143]
There are times when our users (such as me) have travels on a route only near an exit ramp or intersection.  It would be nice to claim those routes as "traveled" without having to "round up" to the nearest adjacent waypoint and claiming more mileage than actually traveled.

**Comment by michihdeu** [comment]
Got it: http://forum.travelmapping.net/index.php?topic=2758.msg12310#msg12310
> There are a number of routes I've only driven at all because I made a gas or food stop right off an exit ramp.  Not nearly enough to "round up" to a nearby TM waypoint.
Zero-length segments

**Comment by jteresco** [comment]
I haven't thought it through and am unlikely to do anything about it any time soon anyway, but a couple of possibilities:

1) A new format of line that means less than 1 entire segment traveled. (I think not hard)
2) A line that repeats the closest waypoint as start and end. (I think easiest and has the added benefit that you're indicating at least to yourself where you traveled a tiny piece)
3) Mark as traveled for any list entry for a route that doesn't result in a valid set of segments being marked.

**Comment by yakra** [comment]
Of these options, I think # 2 makes the most sense.
Not incredibly wild about including it, but hey. \*shrug\*
How might something like this be handled in siteupdate? Travels are added on a HighwaySegment basis, right? I have trouble seeing how this would fit into the existing framework, absent some significant modifications.

**Comment by michihdeu** [comment]
> I have trouble seeing how this would fit into the existing framework, absent some significant modifications.

Yep, that's also what I don't understand. And is it really an improvement? A new list file format is a big thing and I think the signal-to-noise ratio is quite bad.

Why not just adding user requested wps?
Maybe a new wp format like $Jim_BlaRd. "$" could indicate that it doesn't show up in HB if we wanna "hide" the wps. If a 2nd user needs the wp, it should get an official unhidden wp.

**Comment by jteresco** [comment]
I don't think it's too hard. If it turned out to be hard and resulted in far-reaching changes, I'd probably not bother.  Ultimately, there would need to be a DB entry in ClinchedRoutes and ClinchedConnectedRoutes to indicate that a user has traveled a route.  Right now, those are only added if the user has positive mileage, which only comes from one or more clinched segments.  Maintaining one extra flag to indicate that we should create an entry in those DB tables for this user might be all we need.  Front end stats just come from the DB, and while it might look odd to see a route "traveled" with 0 mileage, that's essentially the situation I'm trying to capture.
